src/apps/pulse/components/Onboarding/PreviewTopUp.tsx

```diff
@@ -0,0 +1,784 @@
+      steps.push({
+        icon: (
+          <img
+            src={OnboardingSwapIcon}
+            alt="Swap to USDC"
+            className="w-9 h-9"
+          />
+        ),
+        title: 'Swap to USDC',
+        description: `$${parseFloat(amount).toFixed(2)} ${selectedToken.symbol} for ${usdcReceived.toFixed(2)} USDC`,
+        bgColor: 'bg-[rgba(75,144,255,0.1)]',
+      });
+    }
+
+    // Step 4: Deposit to Gas Tank
+    const depositAmount = isSelectedTokenUSDC()
+      ? parseFloat(amount)
+      : allocateAmount;
+    steps.push({
+      icon: (
+        <img src={OnboardingGasTankIcon} alt="Gas Tank" className="w-9 h-9" />
+      ),
+      title: 'Deposit to Gas Tank',
+      description: `$${depositAmount.toFixed(2)} USDC into Gas Tank`,
+      bgColor: 'bg-[rgba(255,54,108,0.1)]',
+    });
+
+    return steps;
+  };
+
+  const steps = getSteps();
+
+  // If showing transaction status, render TransactionStatus component
+  if (showTransactionStatus && selectedToken) {
+    return (
+      <div className="w-full max-w-[358px] mx-auto flex items-center justify-center">
+        <TransactionStatus
+          closeTransactionStatus={closeTransactionStatus}
+          userOpHash={userOpHash}
+          chainId={selectedToken.chainId}
+          gasFee={transactionGasFee}
+          isBuy={false}
+          sellToken={{
+            name: selectedToken.name,
+            symbol: selectedToken.symbol,
+            logo: selectedToken.logo,
+            address: selectedToken.address,
+          }}
+          buyToken={null}
+          tokenAmount={amount}
+          sellOffer={sellOffer}
+          payingTokens={payingTokens}
+          usdAmount={amount}
+          useRelayBuy={false}
+          currentStatus={currentTransactionStatus}
+          errorDetails={errorDetails}
+          submittedAt={submittedAt}
+          pendingCompletedAt={pendingCompletedAt}
+          blockchainTxHash={blockchainTxHash}
+          resourceLockTxHash={undefined}
+          completedTxHash={undefined}
+          completedChainId={undefined}
+          resourceLockChainId={undefined}
+          resourceLockCompletedAt={undefined}
+          isResourceLockFailed={false}
+          fromChainId={selectedToken.chainId}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <div className="w-full max-w-[358px] mx-auto">
+      <div className="w-full flex flex-col bg-[#1E1D24] rounded-2xl p-0 relative">
+        {/* Header */}
+        <div className="flex items-center justify-between p-3">
+          <button
+            onClick={onBack}
+            type="button"
+            className="text-white flex items-center justify-center"
+            aria-label="Go back"
+          >
+            <svg width="12" height="12" viewBox="0 0 12 12" fill="none">
+              <path
+                d="M6 12L0 6L6 0"
+                stroke="white"
+                strokeWidth="2"
+                fill="none"
+              />
```

Comment:
Can we convert this to SVG files for consistency like the other files

Reply:
okay